twisted/internet/iocpreactor/abstract.py

```diff
@@ -236,8 +237,9 @@ def doWrite(self):
         if len(self.dataBuffer) - self.offset < self.SEND_LIMIT:
             # If there is currently less than SEND_LIMIT bytes left to send
             # in the string, extend it with the array data.
-            self.dataBuffer = (buffer(self.dataBuffer, self.offset) +
-                               "".join(self._tempDataBuffer))
+            writeView = memoryview(self.dataBuffer)
+            self.dataBuffer = (writeView[self.offset:].tobytes() +
```

Comment:
If I remove memoryview.tobytes() from the places that you recommended, then
when I run the tests, some of them start hanging, such as this one:

```
twisted.internet.test.test_tcp
  AbortConnectionTests_IOCPReactorTests
    test_clientAbortsConnectionThenLosesConnection ...                  [ERROR]
    test_clientAbortsConnectionTwice ...                                [ERROR]
    test_dataReceivedAbort ...                                                [ERROR]
```